lib/money/money.rb

```diff
@@ -137,13 +137,8 @@ def -(other)
   end
 
   def *(numeric)
-    unless numeric.is_a?(Numeric)
-      if Money.config.legacy_deprecations
-        Money.deprecate("Multiplying Money with #{numeric.class.name} is deprecated and will be removed in the next major release.")
-      else
-        raise ArgumentError, "Money objects can only be multiplied by a Numeric"
-      end
-    end
+    raise ArgumentError, "Money objects can only be multiplied by a Numeric" unless numeric.is_a?(Numeric)
```

Comment:
The money multiplication feature was deprecated years ago, so I removed it to avoid having to maintain test coverage for it in this PR with this change.

Reply:
confirmed, we're not seeing this in core
https://app.bugsnag.com/shopify/shopify-production/errors?filters[event.since]=all&filters[error.status]=open&filters[event.class]=ActiveSupport%3A%3ADeprecationException&filters[event.message]=Multiplying%20Money